os/kernel/debug/mem_leak_checker.c

```diff
@@ -52,46 +52,66 @@ struct alloc_node_info_s {
 	struct alloc_node_info_s *next;
 };
 
-static struct alloc_node_info_s *hash_table[HASH_SIZE];
-static struct alloc_node_info_s node_info[MAX_ALLOC_COUNT];
-static struct mem_leak_checker_info_s leak_checker;
-static int checkerpid;
+static struct alloc_node_info_s **g_hash_table;
+static struct alloc_node_info_s *g_node_info;
 
-#ifdef CONFIG_MM_KERNEL_HEAP
-extern struct mm_heap_s g_kmmheap[CONFIG_KMM_NHEAPS];
-#endif
-
-static void hash_init(void)
+static int hash_init(void)
 {
 	int index;
+
+	DEBUGASSERT(g_hash_table == NULL && g_node_info == NULL);
```

Comment:
Hmmmm. Is it necessary to be asserted? If it's not a null, then just return ok? Then, is there any problem?
Is there any chance to execute mem leak twice without end?

Reply:
yes It is checking already in run_mem_leak_checker function.
we don't need this.
```
int run_mem_leak_checker(int checker_pid, char *bin_name)
	if (g_hash_table || g_node_info) {
		printf("mem_leak_checker is already running.\n");
		return ERROR;
	}
```